packages/components/src/modifiers/hds-code-editor.ts

```diff
@@ -0,0 +1,248 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Modifier from 'ember-modifier';
+import { assert, warn } from '@ember/debug';
+import { registerDestructor } from '@ember/destroyable';
+import { task } from 'ember-concurrency';
+import config from 'ember-get-config';
+
+// hds-dark theme
+import hdsDarkTheme from './hds-code-editor/themes/hds-dark-theme.ts';
+import hdsDarkHighlightStyle from './hds-code-editor/highlight-styles/hds-dark-highlight-style.ts';
+
+import type { HdsCodeEditorLanguages } from './hds-code-editor/types.ts';
+import type { ArgsFor, PositionalArgs, NamedArgs } from 'ember-modifier';
+import type { StreamLanguage, StreamParser } from '@codemirror/language';
+import type { Extension } from '@codemirror/state';
+import type { EditorView, ViewUpdate } from '@codemirror/view';
+
+export interface HdsCodeEditorSignature {
+  Args: {
+    Named: {
+      language?: HdsCodeEditorLanguages;
+      value?: string;
+      onInput?: (newVal: string) => void;
+      onBlur?: (editor: EditorView, event: FocusEvent) => void;
+      onSetup?: (editor: EditorView) => unknown;
+    };
+  };
+}
+
+async function defineStreamLangugage(streamParser: StreamParser<unknown>) {
```

Comment:
typo (needs update in a few places)
```suggestion
async function defineStreamLanguage(streamParser: StreamParser<unknown>) {
```

Reply:
😬 🙇